core/services/directrequestocr/drlistener.go

```diff
@@ -23,6 +25,77 @@ import (
 var (
 	_ log.Listener   = &DRListener{}
 	_ job.ServiceCtx = &DRListener{}
+
+	sizeBuckets = []float64{
+		1024,
+		1024 * 4,
+		1024 * 8,
+		1024 * 16,
+		1024 * 64,
+		1024 * 256,
+	}
+
+	promOracleEvent = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_oracle_event",
+		Help: "Metric to track received oracle events",
+	}, []string{"oracle", "event"})
+
+	promRequestInternalError = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_internal_error",
+		Help: "Metric to track internal errors",
+	}, []string{"oracle"})
+
+	promRequestComputationError = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_computation_error",
+		Help: "Metric to track computation errors",
+	}, []string{"oracle"})
+
+	promRequestComputationSuccess = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_computation_success",
+		Help: "Metric to track number of computed requests",
+	}, []string{"oracle"})
+
+	promRequestTimeout = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_timeout",
+		Help: "Metric to track number of timed out requests",
+	}, []string{"oracle"})
+
+	promRequestConfirmed = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_confirmed",
+		Help: "Metric to track number of confirmed requests",
+	}, []string{"oracle", "responseType"})
+
+	promRequestDataSize = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "functions_request_data_size",
+		Help:    "Metric to track request data size",
+		Buckets: sizeBuckets,
+	}, []string{"oracle"})
+
+	promComputationResultSize = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "functions_request_computation_result_size",
+		Help:    "Metric to track computation result size in bytes",
+		Buckets: sizeBuckets,
```

Comment:
Computation result and error are max 256 bytes so they'll all fall into the first bucket.

Reply:
Changed to Gauge for these two. Will try to get min/max/avg.